src/bootstrap/compile.rs

```diff
@@ -676,18 +678,28 @@ impl Step for Rustc {
             } else {
                 false
             }
-        } else if let Some(path) = &builder.config.rust_profile_use {
-            if compiler.stage == 1 {
-                cargo.rustflag(&format!("-Cprofile-use={}", path));
-                cargo.rustflag("-Cllvm-args=-pgo-warn-missing-function");
+        } else if let Some(path) = &builder.config.llvm_profile_generate {
```

Comment:
why are `rust_profile_generate` and `llvm_profile_generate` mutually exlusive?

Reply:
From my understanding this function is only responsible for building Rust crates, not the LLVM libraries.  If `llvm_profile_generate` is set, the code that handles building LLVM will take care of that and then this code will compile the crates appropriately.

This code now makes the original check to see if Rust instrumentation was requested.  If not, it will still instrument the Rust crates if the LLVM libraries are instrumented and linkage is set to `static`, as a way of ensuring that the necessary runtime libraries are present.